.github/workflows/update-dependencies-from-metadata.yml

```diff
@@ -0,0 +1,288 @@
+name: Update Dependencies From Metadata (Retrieve, Metadata, Compile, Test, Create PR)
+
+on:
+  workflow_dispatch:
+  # https://crontab.guru/every-12-hours
+  schedule:
+    - cron: '0 */12 * * *'
+
+jobs:
+  select-go-version:
+    name: Select Go Version
+    runs-on: ubuntu-latest
+    outputs:
+      go-version: ${{ steps.select-go-version.outputs.go-version }}
+    steps:
+      - name: Select Go Version
+        id: select-go-version
+        run: echo "::set-output name=go-version::>=1.18.0"
+
+  retrieve:
+    name: Retrieve New Versions and Generate Metadata
+    needs:
+      - select-go-version
+    runs-on: ubuntu-latest
+    outputs:
+      metadata-filepath: ${{ steps.retrieve.outputs.metadata-filepath }}
+      metadata-json: ${{ steps.retrieve.outputs.metadata-json }}
+      id: ${{ steps.retrieve.outputs.id }}
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+
+      - name: Setup Go '${{ needs.select-go-version.outputs.go-version }}'
+        uses: actions/setup-go@v3
+        with:
+          go-version: ${{ needs.select-go-version.outputs.go-version }}
+
+      - name: Run Retrieve
+        id: retrieve
+        working-directory: dependency
+        run: |
+          OUTPUT="/tmp/metadata.json"
+
+          make retrieve \
+            buildpackTomlPath="${{ github.workspace }}/buildpack.toml" \
+            output="${OUTPUT}"
+
+          id=$(jq -r .[0].id < "${OUTPUT}")
+          content=$(jq -r < "${OUTPUT}")
+
+          content="${content//'%'/'%25'}"
+          content="${content//$'\n'/'%0A'}"
+          content="${content//$'\r'/'%0D'}"
+
+          echo "::set-output name=metadata-filepath::${OUTPUT}"
```

Comment:
I wonder if it makes sense to switch to the [environment files](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files) paradigm for setting output now rather than later, especially since these are new workflows. GHAs won't fully remove `set-output` support until May 2023 so I can see the argument for leaving things as is for a later concerted effort, but seems like fairly low-hanging fruit.

Reply:
If we do this, lets make the change in the upstream github-config workflow!

Reply:
Agreed, let's move the discussion to `github-config` as part of https://github.com/paketo-buildpacks/github-config/pull/604